packages/fluence-js/src/internal/FluencePeer.ts

```diff
@@ -700,31 +718,12 @@ export class FluencePeer {
         const particleId = req.particleContext.particleId;
 
         if (this._fluenceAppService && this._marineServices.has(req.serviceId)) {
-            const args = JSON.stringify(req.args);
-            const rawResult = await this._fluenceAppService.callService(req.serviceId, req.fnName, args, undefined);
+            const result = await this._fluenceAppService.callService(req.serviceId, req.fnName, req.args, undefined);
 
-            try {
-                const result = JSON.parse(rawResult);
-                if (typeof result.error === 'string' && result.error.length > 0) {
-                    return {
-                        retCode: ResultCodes.error,
-                        result: result.error,
-                    };
-                }
-
-                if (result.result === undefined) {
-                    throw new Error(
-                        `Call to marine-js returned no error and empty result. Original request: ${jsonify(req)}`,
-                    );
-                }
-
-                return {
-                    retCode: ResultCodes.success,
-                    result: result.result,
-                };
-            } catch (e) {
-                throw new Error(`Call to marine-js. Result parsing error: ${e}, original text: ${rawResult}`);
-            }
+            return {
+                retCode: ResultCodes.success,
+                result: result as JSONValue,
+            };
```

Comment:
I think in the furture we should get rid of the typecasts because who knows what data can be sent by bad actors - I think everything should be checked and acted apon appropriatly instead of just failing